tests/manager_tests/test_manager.py

```diff
@@ -493,6 +502,15 @@ def test_get_update_delete_module(self):
                                       num_args=1)
         self.assertEqual(args[0], {"name": "module1"})
 
+    def test_generate_dashboard(self):
+        manager = TxManager()
+        dashboard = manager.generate_dashboard()
+        title_expected = 'Tx-Manager Dashboard'
+        message_expected = '<h1>TX-Manager Dashboard</h1><h2>Module Attributes</h2><br><table>\n            <tr><td class="hdr" colspan="2">module3</td></tr>\n            <tr><td class="lbl" >Type:</td><td>conversion</td></tr>\n            <tr><td class="lbl" >Input Format:</td><td>md</td></tr>\n            <tr><td class="lbl" >Output Format:</td><td>html</td></tr>\n            <tr><td class="lbl" >Resource Types:</td><td>other, yet_another</td></tr>\n            <tr><td class="lbl" >Version:</td><td>1</td></tr>\n            <tr><td class="lbl" >Private Links:</td><td>{0}/tx/private/md2html</td></tr>\n            <tr><td class="lbl" >Public Links:</td><td>https://api.example.com/tx/convert/md2html</td></tr>\n            <tr><td class="hdr" colspan="2">module2</td></tr>\n            <tr><td class="lbl" >Type:</td><td>conversion</td></tr>\n            <tr><td class="lbl" >Input Format:</td><td>usfm</td></tr>\n            <tr><td class="lbl" >Output Format:</td><td>html</td></tr>\n            <tr><td class="lbl" >Resource Types:</td><td>ulb</td></tr>\n            <tr><td class="lbl" >Version:</td><td>1</td></tr>\n            <tr><td class="lbl" >Options:</td><td>{"pageSize": "A4"}</td></tr>\n            <tr><td class="lbl" >Public Links:</td><td>https://api.example.com/tx/convert/usfm2html</td></tr>\n            <tr><td class="hdr" colspan="2">module1</td></tr>\n            <tr><td class="lbl" >Type:</td><td>conversion</td></tr>\n            <tr><td class="lbl" >Input Format:</td><td>md</td></tr>\n            <tr><td class="lbl" >Output Format:</td><td>html</td></tr>\n            <tr><td class="lbl" >Resource Types:</td><td>obs, ulb</td></tr>\n            <tr><td class="lbl" >Version:</td><td>1</td></tr>\n            <tr><td class="lbl" >Options:</td><td>{"pageSize": "A4"}</td></tr>\n            <tr><td class="lbl" >Public Links:</td><td>https://api.example.com/tx/convert/md2html</td></tr>\n        </table>'
```

Comment:
This looks really fragile. It would be best to check for specific values in specific HTML tags rather than trying to match the entire HTML string, with all its white space and new lines.